pulsar-client-api/src/main/java/org/apache/pulsar/client/api/Message.java

```diff
@@ -81,7 +81,8 @@
      *
      * <p>Only messages received from the consumer will have a message id assigned.
      *
-     * @return the message id null if this message was not received by this client instance
+     * @return the message id
+     * @throws NullPointerException if the message is null (message was not received by this client instance)
```

Comment:
I think we should really fix the implementation instead of throwing exception

Reply:
@merlimat I thought about doing that, but that assumes making sure that null check is added where the getMessageId() is used (which is impossible that the code outside of pulsar repo). 
I can also simply remove checkNotNull and simply let the NPE happen eventually someplace else but I'd love to have more than one voice for this option.

Reply:
Seems returning null or throwing NPE is really opinion based. I think it is important not to break client code, e.g. if client code is checking for NPE and not doing null check then change to return null could possible cause NPE in their program.
And I think we should also keep it consistent with other code in terms of returning null or throw.

Reply:
In my opinion is it unlikely that a Java program catches an NPE in order to detect this case.


We can remove the null check and release the change in 2.8.0.
I am pretty sure that no one will notice this change